packages/codegen-ui-react/lib/forms/form-renderer-helper.ts

```diff
@@ -480,12 +506,67 @@ export function buildOnBlurStatement(fieldName: string) {
   );
 }
 
+/**
+ * if the onChange variable is defined it will send the current state of the fields into the function
+ * the function expects all fields in return
+ * the value for that fields onChange will be used from the return object for validation and updating the new state
+ *
+ *
+ * ex. if the field is email
+ * const returnObject = onChange({ email, ...otherFieldsForForm });
+ * const value = returnObject.email;
+ *
+ * this value is now used in email validation and setting the state
+ */
+export const buildOverrideOnChangeStatement = (
+  fieldName: string,
+  fieldConfigs: Record<string, FieldConfigMetadata>,
+): IfStatement => {
+  return factory.createIfStatement(
+    factory.createIdentifier('onChange'),
+    factory.createBlock(
+      [
+        buildModelFieldObject(true, fieldConfigs),
+        factory.createVariableStatement(
+          undefined,
+          factory.createVariableDeclarationList(
+            [
+              factory.createVariableDeclaration(
+                factory.createIdentifier('result'),
+                undefined,
+                undefined,
+                factory.createCallExpression(factory.createIdentifier('onChange'), undefined, [
+                  factory.createIdentifier('modelFields'),
+                ]),
+              ),
+            ],
+            NodeFlags.Const,
+          ),
+        ),
+        factory.createExpressionStatement(
+          factory.createBinaryExpression(
+            factory.createIdentifier('value'),
+            factory.createToken(SyntaxKind.EqualsToken),
+            factory.createBinaryExpression(
+              buildAccessChain(['result', ...fieldName.split('.')]),
```

Comment:
this is to build the access chain to support nested json (example in snapshot diff and below)
```js
value = result?.bio?.favoriteAnimal ?? value;
```